### ISSUE TITLE
REKO deck card image extractor

### ISSUE THREAD

**Issue by Sembiance** [jsummers/deark#82]
A very popular format for digital playing card graphics back in the day was [REKO](http://fileformats.archiveteam.org/wiki/REKO_Cardset)
There is a website that collects the cardsets and has some info about the format: https://www.rekonet.org/index.html

I found some code out there that can extract the card images, converting to PPM: https://github.com/Sembiance/reko2png/blob/master/rekotoppm.c

However I have encountered some reko files that do not work with it: https://discmaster.textfiles.com/browse/705/The%20Games%20Room.iso/amiga/card_games/klondike-iii/cards
On the above link, at the very bottom you'll notice 5 reko decks that didn't convert.
Looking at [claudio_cards.reko](https://discmaster.textfiles.com/view/705/The%20Games%20Room.iso/amiga/card_games/klondike-iii/cards/claudia_cards.reko) it looks like it has 90x132 cards but the code expects 88x130. Just changing the code to try with any width/height didn't work.

I'm not aware of any other code/program that can convert these.
There are hundreds of unique decks on [discmaster](https://discmaster.textfiles.com/search?formatid=rekoCardset&dedup=dedup)

It would be great if deak could add support for extracting these, especially the new variants I found.

EDIT: Found another variant: [intro.reko](https://dev.discmaster2.textfiles.com/view/705/The%20Games%20Room.iso/amiga/card_games/klondike-iii/pics/intro.reko)

**Comment by jsummers** [comment]
Thanks for the idea. I'm interested, but I haven't decided whether to work on it.

I'm hoping to find some more documentation, or compatibly-licensed source code -- though I may be able to do without it. (I'm not above looking at GPL code to answer a question or two about a format, but I'm trying to be careful. I can't use such code in Deark.)

**Comment by Sembiance** [comment]
I found some old amiga code that parses the format and has a format description: https://discmaster.textfiles.com/view/4782/Amiga%20Plus%20CD%20-%202004%20-%20No.%2006.iso/AmigaPlus/Tools/mOS_Pegasos/EssentialPack1/sdiREKO-DT.lha/DataFormat.TXT

According to the readme and the source code files it's `freeware`

**Comment by jsummers** [comment]
I've implemented REKO support. Unfortunately, the uncommon "RKP 24" format is handled differently than the other formats -- each card is extracted to its own file.

I suspect intro.reko is some sort of resource file used by the software, not a cardset.

The other problematic files are probably bad, mainly in that the card width is labeled incorrectly. They work well enough if I assume the width is always 88 pixels, so I've added a hack to do that. So far, it doesn't seem to break anything.
